packages/block-editor/src/components/block-tools/insertion-point.js

```diff
@@ -178,10 +193,10 @@ function InsertionPointPopover( {
 		}
 	}
 
-	// Only show the inserter when there's a `nextElement` (a block after the
-	// insertion point). At the end of the block list the trailing appender
-	// should serve the purpose of inserting blocks.
-	const showInsertionPointInserter = nextElement && isInserterShown;
+	// Only show the in-between inserter between blocks, so when there's a
+	// previous and a next element.
+	const showInsertionPointInserter =
+		previousElement && nextElement && isInserterShown;
```

Comment:
`isInserterShown` might suffice here. The component controls it is the in between inserter hook showing or hiding the inserter.